aws-cpp-sdk-transfer/source/transfer/UploadFileRequest.cpp

```diff
@@ -638,17 +614,12 @@ void UploadFileRequest::AddCompletedPart(PartRequestRecord& partRequest, const A
 
     if (m_completedParts.size() == GetTotalParts() && !IsDone())
     {
+        lockPart.unlock();

```

Comment:
Same as above, not sure what this lock protects. Is it ok to call CompleteUpload with no lock?